src/components/StatusIndicator/_mixins.scss

```diff
@@ -4,102 +4,114 @@
+    list-style: none;
   }
 
-  /// Styles for elements in each step.
   &__step {
-    width: 100%;
-    margin: $carbon--spacing-02 0;
+    /// Error color.
+    /// @type Color
+    $status-indicator--error__color: $support-01;
 
+    &__icon-wrapper,
     &-wrapper {
       display: flex;
-      width: 100%;
       align-items: center;
-    }
 
-    &__label,
-    .#{$loading}__text {
-      @include carbon--type-scale($step: 2);
-      color: $text-01;
-      margin: 0 $carbon--spacing-03;
+      > .#{$prefix}--inline-loading__text {
+        color: $body__color__text;
+      }
     }
 
-    &__icon,
-    .#{$loading}__svg {
-      width: carbon--mini-units(2.5);
-      height: carbon--mini-units(2.5);
+    &__icon-wrapper {
+      /// Wrapper dimensions.
+      /// @type Length
+      $status-indicator__wrapper__sizing__dimensions: carbon--mini-units(
+        $count: 4,
+      );
+
+      width: $status-indicator__wrapper__sizing__dimensions;
+      height: $status-indicator__wrapper__sizing__dimensions;
+      justify-content: center;
     }
 
-    &--incomplete #{$step}__label {
-      color: $disabled-03;
+    &__icon {
+      /// Icon dimensions.
+      /// @type Length
+      $status-indicator__icon__sizing__dimensions: carbon--mini-units(
+        $count: 2,
+      );
+
+      width: $status-indicator__icon__sizing__dimensions;
+      height: $status-indicator__icon__sizing__dimensions;
     }
 
-    &--incomplete #{$step}__icon {
-      fill: $disabled-03;
+    &__label {
+      @include carbon--type-style($name: body-short-01);
+      margin-top: 0;
+      margin-bottom: 0;
     }
 
-    &__icon-wrapper {
-      position: relative;
-      display: flex;
-      justify-content: center;
-      align-items: center;
-      width: $carbon--layout-03;
-      height: $carbon--layout-03;
+    &--failed__icon {
+      fill: $status-indicator--error__color;
     }
 
-    &--failed__icon {
-      fill: $support-01;
+    &--incomplete {
+      /// Incomplete color.
+      /// @type Color
+      $status-indicator__icon--incomplete__color: $disabled-03;
+
+      &__icon {
+        fill: $status-indicator__icon--incomplete__color;
+      }
+
+      &__label {
+        color: $status-indicator__icon--incomplete__color;
+      }
     }
 
     &__error-message {
-      @include carbon--type-scale($step: 1);
+      @include carbon--type-style($name: label-01);
+
       display: inline-block;
-      color: $support-01;
-      margin-left: $carbon--spacing-08;
       margin-top: 0;
+      margin-bottom: $status-indicator__spacing__margin;
+      margin-left: $carbon--spacing-07;
+      color: $status-indicator--error__color;
     }
   }
 
-  /// The "Retry" ghost button.
-  &__retry-button {
-    justify-content: left;
-    align-items: left;
-    padding: 0;
-
-    &__icon {
-      fill: $interactive-04;
-    }
-
-    &__message {
-      @include carbon--type-scale($step: 2);
-      margin-left: $carbon--spacing-03;
-    }
+  &__button {
```

Comment:
Would like to still call this `retry-button` instead, if possible? (see prev comments about class names)